environments.md

```diff
@@ -0,0 +1,5 @@
+# Environments
+
+OSM can be installed in any Kubernetes cluster running in a variety of environments such as AWS, Azure Kubernetes Service, Google Cloud, and even bare metal.
+
+In the specific case of Azure Kubernetes Service, [a supported add-on](https://docs.microsoft.com/azure/aks/open-service-mesh-about) is provided to handle installation and configuration of OSM.
```

Comment:
Should we add a disclaimer regarding the addon not being managed by the upstream OSS project but rather by Microsoft?